src/Service/Bulk/BulkCreateUsersService.php

```diff
@@ -0,0 +1,294 @@
+                        : '';
+
+                    $csvData[] = [$username, $userPassword,$userGroupId];
+                    $assignmentCsvData[] = [$username, $lineSlugs];
+
+                    $slugTotalUsersCreated++;
+                }
+
+                $this->csvWriter->writeCsvData(
+                    $lineSlugs,
+                    $prefix,
+                    $csvPath,
+                    $csvFilename,
+                    $automateCsvPath,
+                    $csvData,
+                    $assignmentCsvData
+                );
+
+                $slugWiseTotalUsersArray[$lineSlugs] = array_key_exists($lineSlugs, $slugWiseTotalUsersArray)
+                    ? $slugWiseTotalUsersArray[$lineSlugs] +  $slugTotalUsersCreated
+                    : $slugTotalUsersCreated;
+            }
+        }
+
+        return $slugWiseTotalUsersArray;
+    }
+
+    private function createUserPassword(): string
+    {
+        return substr(str_shuffle('0123456789abcdefghijklmnopqrstvwxyz'), 0, 8);
+    }
+
+    private function getFindLineItemCriteria(): FindLineItemCriteria
+    {
+        $criteria = new FindLineItemCriteria();
+
+        if (!empty($this->lineItemIds)) {
+            $criteria->addLineItemIds(...$this->lineItemIds);
+        }
+
+        if (!empty($this->lineItemSlugs)) {
+            $criteria->addLineItemSlugs(...$this->lineItemSlugs);
+        }
+
+        return $criteria;
+    }
+
+    /**
+     * Function to get all slug data
+     * @throws LineItemNotFoundException
+     */
+    private function getAllLineItemSlugs(): void
+    {
+        $lineItems = $this->lineItemRepository->findAllAsCollection()->jsonSerialize();
+        if (empty($lineItems)) {
+            throw new LineItemNotFoundException('No line items were found in database.');
+        }
+
+        $this->generateSlugData($lineItems);
+    }
+
+    private function getLastUserAssignedToLineItems(array $lineItemSlugs): array
+    {
+        $userNameIncrementArray = [];
+
+        foreach ($lineItemSlugs as $slugKey => $slug) {
+            $assignment = $this->assignmentRepository->findByLineItemId($slugKey);
+            $userNameIncrementArray[$slug] = self::DEFAULT_USERNAME_INCREMENT_VALUE;
+
+            if (!empty($assignment)) {
+                $userData = $assignment->getUser()->getUsername();
+                $userNameArray = explode('_', (string) $userData);
+                $userNameLastIndex = self::DEFAULT_USERNAME_INCREMENT_VALUE;
+                $index = end($userNameArray);
+                if (count($userNameArray) > 0 && is_numeric($index)) {
+                    $userNameLastIndex = (int) $index;
+                }
+                $userNameIncrementArray[$slug] = $userNameLastIndex;
+            }
+        }
+
+        return $userNameIncrementArray;
+    }
+
+    private function createUserGroupId(array $userGroupIds, int $userGroupAssignCount): string
```

Comment:
The same problem as the one raised about method naming above by Sergii.
This method shouldn't be named `createSmth` as it's creating nothing. It should be either `getSmth` or `findSmth` or maybe you have a more suitable naming ;)